Source/Core/Core/IOS/ES/ES.cpp

```diff
@@ -759,6 +764,128 @@ bool ES::IsActiveTitlePermittedByTicket(const u8* ticket_view) const
       Common::swap32(ticket_view + offsetof(IOS::ES::TicketView, permitted_title_id));
   return title_identifier && (title_identifier & ~permitted_title_mask) == permitted_title_id;
 }
+
+bool ES::IsIssuerCorrect(VerifyContainerType type, const IOS::ES::CertReader& issuer_cert) const
+{
+  switch (type)
+  {
+  case VerifyContainerType::TMD:
+    return issuer_cert.GetName().compare(0, 2, "CP") == 0;
+  case VerifyContainerType::Ticket:
+    return issuer_cert.GetName().compare(0, 2, "XS") == 0;
+  case VerifyContainerType::Device:
+    return issuer_cert.GetName().compare(0, 2, "MS") == 0;
+  default:
+    return false;
+  }
+}
+
+ReturnCode ES::WriteNewCertToStore(const IOS::ES::CertReader& cert)
+{
+  const std::string store_path = Common::RootUserPath(Common::FROM_SESSION_ROOT) + "/sys/cert.sys";
+  File::IOFile store_file{store_path, "a+b"};
+  if (!store_file)
+    return ES_EIO;
+
+  // Read the current store to determine if the new cert needs to be written.
+  std::vector<u8> certs_bytes(store_file.GetSize());
+  if (!store_file.ReadBytes(certs_bytes.data(), certs_bytes.size()))
+    return ES_SHORT_READ;
+
+  const std::map<std::string, IOS::ES::CertReader> certs = IOS::ES::ParseCertChain(certs_bytes);
+  // The cert is already present in the store. Nothing to do.
+  if (certs.find(cert.GetName()) != certs.end())
+    return IPC_SUCCESS;
+
+  store_file.Seek(0, SEEK_END);
```

Comment:
Didn't we want to document this `Seek` in light of the discussion on IRC, where it wouldn't append despite specifying `a+b` as mode?